Source/cryptalgo/SecureSocketPort.h

```diff
@@ -107,6 +114,7 @@ namespace Crypto {
 
             // Signal a state change, Opened, Closed or Accepted
             void StateChange() override {
+//                ASSERT(_context != nullptr && _ssl != nullptr);
```

Comment:
Interesting that this ASSERT fires... Maybe during the Close Sequence? Please investigate and remove the assert or make sure that under normal conditions it does not fire (IsClosed() == false) || ((_context != nullptr) && (_ssl != nullptr)));

Reply:
This is interesting. In a previous draft PR you did not wanted such solution but promoted the use of ASSERT.  Update() already includes a check and ASSERT.